base/shell/explorer/trayclock.cpp

```diff
@@ -388,6 +389,38 @@ VOID CTrayClockWnd::UpdateWnd()
             GetParent().SendMessage(WM_NOTIFY, 0, (LPARAM) &nmh);
         }
     }
+
+    const int iDateLength = GetDateFormat(LOCALE_USER_DEFAULT,
+                                          DATE_LONGDATE,
+                                          &LocalTime,
+                                          NULL,
+                                          NULL,
+                                          0);
+    if (iDateLength <= 0)
+    {
+        return;
+    }
+
+    const LPWSTR szDate = reinterpret_cast<LPWSTR>(
```

Comment:
"const" not needed. Also since this is C++, why don't you use `new WCHAR[iDateLength]` instead?